src/Compilers/Core/Portable/InternalUtilities/ISetExtensions.cs

```diff
@@ -17,6 +18,17 @@ public static bool AddAll<T>(this ISet<T> set, IEnumerable<T> values)
             return result;
         }
 
+        public static bool AddAll<T>(this ISet<T> set, ImmutableArray<T> values)
+        {
+            var result = false;
+            foreach (var v in values)
+            {
+                result |= set.Add(v);
```

Comment:
If a set returns `true` only if the element was successfully added, wouldn't you expect `AddAll` to only return true if all elements were successfully added?